seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/checkpoint/CheckpointCoordinator.java

```diff
@@ -827,6 +827,20 @@ public boolean isCompleted() {
                 && !latestCompletedCheckpoint.isRestored();
     }
 
+    public boolean isNoErrorCompleted() {
+        if (latestCompletedCheckpoint == null) {
+            return false;
+        }
+        return latestCompletedCheckpoint.getCheckpointType().isFinalCheckpoint()
+                && (runningJobStateIMap
+                                .get(checkpointStateImapKey)
+                                .equals(CheckpointCoordinatorStatus.FINISHED)
+                        || runningJobStateIMap
+                                .get(checkpointStateImapKey)
+                                .equals(CheckpointCoordinatorStatus.SUSPEND))
```

Comment:
How about extracting this part as a variable？
> runningJobStateIMap.get(checkpointStateImapKey)

Reply:
done